components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14060,6 +14062,273 @@ public Label updateLabel(Label label) throws APIManagementException {
+                }
+                addGatewayVhosts(conn, id, environment.getVhosts());
+                conn.commit();
+            } catch (SQLException e) {
+                conn.rollback();
+                handleException("Failed to add VHost: " + uuid, e);
+            }
+        } catch (SQLException e) {
+            handleException("Failed to add VHost: " + uuid, e);
+        }
+        return environment;
+    }
+
+    /**
+     * Add VHost assigned to gateway environment
+     *
+     * @param connection connection
+     * @param id Environment ID in the databse
+     * @param vhosts list of VHosts assigned to the environment
+     * @throws APIManagementException if falied to add VHosts
+     */
+    private void addGatewayVhosts(Connection connection, int id, List<VHost> vhosts) throws
+            APIManagementException {
+        try (PreparedStatement prepStmt = connection.prepareStatement(SQLConstants.INSERT_GATEWAY_VHOSTS_SQL)) {
+            for (VHost vhost : vhosts) {
+                prepStmt.setInt(1, id);
+                prepStmt.setString(2, vhost.getHost());
+                prepStmt.setString(3, vhost.getHttpContext());
+                prepStmt.setString(4, vhost.getHttpPort().toString());
+                prepStmt.setString(5, vhost.getHttpsPort().toString());
+                prepStmt.setString(6, vhost.getWsPort().toString());
+                prepStmt.setString(7, vhost.getWssPort().toString());
+                prepStmt.addBatch();
+            }
+            prepStmt.executeBatch();
+        } catch (SQLException e) {
+            handleException("Failed to add VHosts for environment ID: " + id, e);
+        }
+    }
+
+    /**
+     * Delete all VHosts assigned to gateway environment
+     *
+     * @param connection connection
+     * @param id Environment ID in the databse
+     * @throws APIManagementException if falied to delete VHosts
+     */
+    private void deleteGatewayVhosts(Connection connection, int id) throws
+            APIManagementException {
+        try (PreparedStatement prepStmt = connection.prepareStatement(SQLConstants.DELETE_GATEWAY_VHOSTS_SQL)) {
+            prepStmt.setInt(1, id);
+            prepStmt.executeUpdate();
+        } catch (SQLException e) {
+            handleException("Failed to delete VHosts for environment ID: " + id, e);
+        }
+    }
+
+    /**
+     * Whether an Environment exists in the tenant domain with the given name
+     *
+     * @param tenantDomain tenant domain
+     * @param envName environment name
+     * @return whether an Environment exist or not
+     * @throws APIManagementException if failed to find
+     */
+    public boolean isEnvironmentNameExist(String tenantDomain, String envName) throws APIManagementException {
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement prepStmt = connection.prepareStatement(SQLConstants.GET_ENVIRONMENT_NAME_COUNT_BY_TENANT_SQL)) {
```

Comment:
do select 1 query to  check the existence

Reply:
Removed `isEnvironmentNameExist` method, previously used for check dynamic env existence, now checking for all environments.
fixed with https://github.com/wso2/carbon-apimgt/pull/9890/commits/8b56851663554865bad62d7ac549af4a1f58f55b